### ISSUE TITLE
Ghost text appearing after pasting another text

### ISSUE THREAD

**Issue by warcayac** [zsh-users/zsh-autosuggestions#766]
### To Reproduce

When pasting any text after a command, a ghost text (last match) on right appears when pressing the left-arrow key, this ghost text stays there till I press any key like backspace key.

https://github.com/zsh-users/zsh-autosuggestions/assets/7342885/850025f4-3029-4d73-b175-75af7f6a1795

### Expected behavior

No ghost text displaying

### Desktop
 - OS: EndeavourOS Linux x86_64 
 - Kernel: 6.5.9-zen2-1-zen 
 - Shell: zsh 5.9 
 - DE: Plasma 5.27.9 
 - WM: KWin 
 - Terminal: HyperTerm 
 - Terminal Font: JetBrainsMono Nerd Font 
 - Plugin version: c3d4e576c9c86eac62884bd47c01f6faed043fc5

**Comment by TobiPeterG** [comment]
I have the same issue on openSUSE Tumbleweed with GNOME.
When inserting text and pressing the right arrow key when previously text was auto suggested, it gets inserted anyway

**Comment by xu-minghao317** [comment]
this is a known issue of ZSH, usually it can be fixed by setting the `$LOCALE` properly.

```shell
echo "export LANG=en_US.UTF-8" >> .zshenv
echo "export export LC_ALL=en_US.UTF-8" >> .zshenv
```

**Comment by jathek** [comment]
> this is a known issue of ZSH, usually it can be fixed by setting the `$LOCALE` properly.
> 
> ```shell
> echo "export LANG=en_US.UTF-8" >> .zshenv
> echo "export LC_ALL=en_US.UTF-8" >> .zshenv
> ```

This doesn't fix the issue for me with autosuggestions c3d4e576c9c86eac62884bd47c01f6faed043fc5 and zsh 5.9 (x86_64-pc-msys) in Windows Terminal

**Comment by xu-minghao317** [comment]
Have you relaunch your terminal after this?

**Comment by jathek** [comment]
Yes, I relaunched after removing "completion" and the ghost text is gone now. So it is something not working with the "completion" strategy.

**Comment by gummikonsumer** [comment]
I am using this plugin locally, and on a linux machine over ssh. it seems to work ok locally, but not on remote machine.

[here is a screen-cap](https://asciinema.org/a/Mg3xQjktPRq9NONqAJijpH7FC)

local .zshrc:

```
export LANG=en_US.UTF-8
export LC_ALL=en_US.UTF-8
ZSH_AUTOSUGGEST_STRATEGY=(match_prev_cmd history)
plugins=(gitfast docker docker-compose gh zsh-autosuggestions fast-syntax-highlighting)
```

remote .zshrc:

```
export LANG=en_US.UTF-8
export LC_ALL=en_US.UTF-8
ZSH_AUTOSUGGEST_STRATEGY=(match_prev_cmd history)
plugins=(gitfast docker docker-compose gh zsh-autosuggestions fast-syntax-highlighting)
```

I don't know if it's related, but additionally it says

```
Error opening terminal: xterm-ghostty.
```

when I try to run `nano` on remote.

**Comment by raylu** [comment]
I also was experiencing a variant of this issue. same symptoms, but I didn't have `completion` strategy on (I had the default of just `history`)

I did have an ancient bracketed paste script that I removed, which fixed it https://gitlab.com/raylu/dotfiles/-/commit/9854077df4a53570c89cd36928e099135155474a (it's built-in as of zsh 5.1)
